elastic/kibana
# [ResponseOps][Actions] Should trim values before validation in UI and API

Resolves [#100770](https://github.com/elastic/kibana/issues/100770#:~:text=UI%20and%20API-,%23100770,-Open)

## Summary

The UI is already trimming values before validation. For the API, I added a function to validate empty strings when a user creates a connector. This is to prevent creating a connector with space or an empty string for a value.

### Checklist

Delete any items that are not applicable to this PR.

### To Verify

- Run a curl command to create a connector. The `service` field is only a space, and it should trim the value and then should throw an error saying that an empty string is not valid.
```
curl -X POST -u $USER:$PASS "$HOST:$PORT/api/actions/connector" -H 'kbn-xsrf: true' -H 'Content-Type: application/json' -d'
{
  "name": "tester-email-connector-type",
   "connector_type_id": ".email",
   "config": {
     "service": " ",    
     "from": "testsender@test.com",
     "host": "validhostname",
     "port": 8080,
     "secure": false
    }
}
```
- You can also run other requests to verify that you can't submit other empty strings.
```
curl -X POST -u $USER:$PASS "$HOST:$PORT/api/actions/connector" -H 'kbn-xsrf: true' -H 'Content-Type: application/json' -d'
{
  "name": " tester-email-connector-type ",
   "connector_type_id": ".email",
   "config": {
     "service": "other", 
     "from": "testsender@test.com",
     "host": "",
     "port": 8080,
     "secure": false
    }
}
```
